src/table/utils/styling-utils.js

```diff
@@ -96,26 +118,26 @@ export function getColumnStyle(styling, qAttrExps, stylingInfo) {
   };
 }
 
-export function getSelectionColors(background = STYLING_DEFAULTS.WHITE, cell, selectionState) {
+export function getSelectionColors(cell, selectionState, background, tableBackgroundColor) {
   const { colIdx, rows, api } = selectionState;
 
   if (api.isModal()) {
-    if (colIdx !== cell.colIdx) {
-      return { background: `${STYLING_DEFAULTS.EXCLUDED_BACKGROUND}, ${background}` };
-    }
+    if (colIdx !== cell.colIdx)
+      return {
+        background: `${STYLING_DEFAULTS.EXCLUDED_BACKGROUND}, ${
+          background || tableBackgroundColor || STYLING_DEFAULTS.WHITE
+        }`,
+      };
 
-    for (let i = 0; i < rows.length; i++) {
-      if (rows[i].qElemNumber === cell.qElemNumber) {
-        return SELECTION_STYLING.SELECTED;
-      }
-    }
+    const match = (row) => row.qElemNumber === cell.qElemNumber;
+    if (rows.some(match)) return SELECTION_STYLING.SELECTED;
```

Comment:
Is this as fast? I used for loops for speed here

Reply:
yes, the array method is faster than for loops. Did a quick search https://leanylabs.com/blog/js-forEach-map-reduce-vs-for-for_of/

Reply:
awesome

Reply:
I think you got to a wrong conclusion from that page. When I read it it states that for loops are almost always better performing than the array methods

Reply:
You are right. I misunderstood it as the x-axis is ops/sec! As the performance is more important here, I gonna revert it.

Reply:
Had a test
<img width="666" alt="Screenshot 2022-03-01 at 19 43 27" src="https://user-images.githubusercontent.com/4471489/156230513-3598ba3f-2aa4-4a16-b6d8-6d13fa288343.png">
<img width="652" alt="Screenshot 2022-03-01 at 19 44 57" src="https://user-images.githubusercontent.com/4471489/156230518-bdc119e6-f134-43d1-b49c-64a3283f94af.png">
for loop is faster